llmfoundry/data/finetuning/dataloader.py

```diff
@@ -257,9 +262,15 @@ def build_finetuning_dataloader(
         dataset_constructor_args = {
             k: v
             for k, v in dataset_cfg.items()
-            if k in dataset_constructor_keys and
-            k not in {'streams', 'packing_ratio'}
+            if k in allowed_dataset_config_keys and k not in {
```

Comment:
Is there a way to do this without hardcoding the expected streaming keys? This requires that we manually keep track of any changes in the streaming repo and sync them to foundry whenever these args change on either side.

Reply:
``streams`` needs to be hardcoded to get excluded. The others may be able to get programatically. But I am also inclined to keep the change minimal, in case it may affect any existing use cases unexpectedly.

I had to add these extra ones because this change [this](https://github.com/mosaicml/llm-foundry/blob/f034a8612708ca7b5c6a0523dbca9f03ffbf28e6/llmfoundry/data/finetuning/tasks.py#L655), breaks a unit test: tests/data/test_packing.py: test_auto_packing_with_streaming_dataloader.

Reply:
Yes, u r right~~ I don't need to change the line.